### PR TITLE
Use Terraform to manage API deployments in oss-vdb-test

### PR DESCRIPTION
**Pull request by michaelkedar**

### CODE REVIEW

File: gcp/api/v1/README.md [hunk 1]
```diff
@@ -41,10 +41,9 @@ To deploy the service proxy,
 
 ### oss-vdb-test Project
 
-```
-../deploy_service_proxy oss-vdb-test osv-grpc-v1-ivqje5d2hq-uc.a.run.app osv-grpc-v1 /path/to/api_config.yaml
-```
+Deployment is handled through terraform.
 
-Configuring the Cloud Endpoint requires the Cloud Run service to already exist before running, as it needs the URL.
+Copy `api_descriptor.pb` into `deployment/terraform/environments/oss-vdb-test/api/`,
```
Comment (oliverchang) [inline gcp/api/v1/README.md:46]: Can we set up a symlink instead?

File: deployment/terraform/modules/osv/scripts/gcloud_build_image [hunk 2]
```diff
@@ -0,0 +1,127 @@
+#!/bin/bash
```
Comment (oliverchang) [inline deployment/terraform/modules/osv/scripts/gcloud_build_image:1]: CAn you link to https://github.com/GoogleCloudPlatform/esp-v2/blob/master/docker/serverless/gcloud_build_image here as well?

File: deployment/terraform/modules/osv/osv_api.tf [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+resource "google_endpoints_service" "grpc_service" {
+  project      = var.project_id
+  service_name = var.api_url
+  grpc_config = templatefile(
+    "api/api_config.tftpl",
+    {
+      service_name = var.api_url,
+      backend_url  = replace(google_cloud_run_service.api_backend.status[0].url, "https://", "grpcs://")
+  })
+  protoc_output_base64 = filebase64("api/api_descriptor.pb")
```
Comment (oliverchang) [inline deployment/terraform/modules/osv/osv_api.tf:38]: If api_descriptor.pb is changed, does this cause the proxy service to be re-deployed as well?

**Review by oliverchang — COMMENTED** [review]
very nice work!!!!!

looks generally good. just some some questions

File: deployment/terraform/modules/osv/variables.tf [hunk 4]
```diff
@@ -12,3 +12,13 @@ variable "vulnerabilities_export_bucket" {
   type        = string
   description = "Name of bucket to export vulnerabilities to."
 }
+
+variable "api_url" {
+  type        = string
+  description = "URL to serve the OSV API on. Domain ownership and DNS settings has to be set up manually."
```
Comment (oliverchang) [inline deployment/terraform/modules/osv/variables.tf:18]: What kind of manual set up is required? Can we have some documentation somewhere to explain this?

File: deployment/terraform/modules/osv/osv_api.tf (hunk 3 above)
Comment (michaelkedar) [reply deployment/terraform/modules/osv/osv_api.tf:38]: Yep - those file functions stores the content of the file in the argument, so changes to the file will be picked up and propagated through to dependent resources.

File: deployment/terraform/modules/osv/variables.tf (hunk 4 above)
Comment (michaelkedar) [reply deployment/terraform/modules/osv/variables.tf:18]: It required verifying ownership of the test.osv.dev domain at https://www.google.com/webmasters/verification/home
and adding a CNAME entry for api.test.osv.dev pointing to ghs.googlehosted.com.

I'll note this in the terraform README

File: deployment/README.md [hunk 5]
```diff
@@ -6,6 +6,26 @@ export GOOGLE_CLOUD_PROJECT=<PROJECT_ID>
...
+    `https://www.google.com/webmasters/verification/verification?authuser=0&domain=custom.domain.name`
+
+    replacing `custom.domain.name` with the domain to be verified.
+    
+    (This link is usually generated when adding a domain mapping to a service in Cloud Run.
+    I don't know how to navigate to that page otherwise. Trying to add a property from
+    [Webmaster Central](https://www.google.com/webmasters/verification/home)
+    adds it as a site, rather than as a domain.)
+
+2. Add DNS CNAME record mapping the API URL to `ghs.googlehosted.com.`
```
Comment (oliverchang) [inline deployment/README.md:27]: Is this a different URL from `custom.domain.name` ? If so, let's make this clearer.

**Review by oliverchang — APPROVED** [review]

**Pull request merged**